src/py/flwr/cli/run/run.py

```diff
@@ -94,61 +73,62 @@ def run(
 
     typer.secho("Success", fg=typer.colors.GREEN)
 
-    if use_superexec:
-        _start_superexec_run(
-            parse_config_args(config_overrides, separator=","), directory
+    try:
+        federation_name = config["flower"]["federations"]["default"]
+        federation = config["flower"]["federations"][federation_name]
+    except KeyError as err:
+        typer.secho(
+            "❌ The project's `pyproject.toml` needs to declare "
+            "a default federation (with a SuperExec address or an "
+            "`options.num-supernodes` value).",
+            fg=typer.colors.RED,
+            bold=True,
         )
-        return
+        raise typer.Exit(code=1) from err
 
-    server_app_ref = config["flower"]["components"]["serverapp"]
-    client_app_ref = config["flower"]["components"]["clientapp"]
+    if "address" in federation:
```

Comment:
Let's move the body of this if/else into two functions:
- `_flwr_run_with_superexec(federation, [...other args...])`
- `_flwr_run_without_superexec(federation, [...other args...])`